### ISSUE TITLE
Why Microsoft Xtra tags are marked as not-writable?

### ISSUE THREAD

**Issue by mnaoumov** [exiftool/exiftool#70]
According to https://exiftool.org/TagNames/Microsoft.html exiftool can't write many Microsoft-specific tags, which especially would be useful to fill Windows file properties

![image](https://user-images.githubusercontent.com/735878/100689014-3d5bfd00-3349-11eb-88ef-7b0a564688d5.png)

I am interested to understand what is the reason behind this limitation? Is it technical limitation? I am interested to try to create a pull request to add this support, but I would like to get some feedback from the author/maintainers first

**Comment by boardhead** [comment]
I don't think it is necessary to write thes.

See [this forum post](https://exiftool.org/forum/index.php?topic=6591.msg32875#msg32875) for a list of tags that Windows reads as the file properties.

- Phil

**Comment by mnaoumov** [comment]
Maybe it's not necessary for image files but it's certainly necessary for **MP4** files. I've created the whole list of the Windows properties attributes that are impossible to set via exiftool for **MP4** files

![image](https://user-images.githubusercontent.com/735878/100691525-89f60700-334e-11eb-9adf-8e51142b0d58.png)

Title: read/write
-QuickTime:Title="aaa"

Subtitle: read only
-Microsoft:Subtitle

Rating: read only, values: (nothing), 1, 25, 50, 75, 99 - correspond to amount of stars (0..5)
-Microsoft:SharedUserRating 

Tags: read only
-Microsoft:Category

Comments:  read/write
-QuickTime:Comment  

![image](https://user-images.githubusercontent.com/735878/100691561-99755000-334e-11eb-81a0-ddd4be26ed4a.png)

Contributing artists:  read/write
-QuickTime:Artist='xxx1'

Year: read/write. # symbol is intentional, without it exiftool requires date in the long form, which doesn't make sense for this field 
- QuickTime:ContentCreateDate#=1492

Genre:  read/write
-QuickTime:Genre='xxx2'

Directors: read only
-Microsoft:Director

Producers: read only
-Microsoft:Producer

Writers: read only
-Microsoft:Writer
 
Publishers: read only
-Microsoft:Publisher

Content provider: read only
-Microsoft:ContentDistributor

Encoded by: read only
-Microsoft:EncodedBy

Author URL: read only
-Microsoft:AuthorURL

Promotion URL: read only
-Microsoft:PromotionURL

Parental rating: read only
-Microsoft:ParentalRating

![image](https://user-images.githubusercontent.com/735878/100691663-c9bcee80-334e-11eb-81ee-c5851f5fb813.png)

Composers:  read/write
-QuickTime:Composer='xxx12'

Conductors:  read only
-Microsoft:Conductor
 
Period:  read only
-Microsoft:Period

Mood:  read only
-Microsoft:Mood

Part of set:  read/write
-QuickTime:DiskNumber='10 of  20'

Initial key:  read only
-QuickTime:InitialKey

Beats-per-minute:  read only
-QuickTime:BeatsPerMinute

**Comment by boardhead** [comment]
Thanks for this list, it is useful.  I'll look into what is involved in writing these tags.

- Phil

**Comment by boardhead** [comment]
It would help to have a sample of all the tags you want to be able to write.  Would it be possible for you to prepare a very small .mp4 containing these tags, and attach it here?

**Comment by mnaoumov** [comment]
[sample.zip](https://github.com/exiftool/exiftool/files/5625157/sample.zip)

**Comment by boardhead** [comment]
Thanks!  This is very useful.  I'll post back here if I have any updates, but it could be a while before I get a chance to work on this.

**Comment by mnaoumov** [comment]
@boardhead I assume, the answer for my original question is "there is no an explicit reason for those tags not to be writable" so I can also try to work on the pull request. I've never done Perl before so I'll also need some time to try to fix the issue on my own

**Comment by boardhead** [comment]
I think you'll find it a very difficult project to tackle, but go ahead if you want.  If you do submit a pull request it may not save me a lot of work because it will probably take more time for me to vet your code than it would to write my own.

- Phil

**Comment by boardhead** [comment]
I've just released 12.20 with the ability to write Microsoft Xtra tags.  (At least all the ones in your sample, plus a few more.)

I just have one comment/question/concern:  I'm very glad that your wrote multiple items for some tags -- I have made these tags List-type so ExifTool will deal with multiple values properly.  Are these all of the tags that support multiple values?  Note that the value of Publisher is "writer1;writer2", which is stored as a single string.  Is this correct?  I don't know why this isn't stored as two separate strings like multiple values of the other tags.
